src/App.tsx

```diff
@@ -1,26 +1,117 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getTodos } from './api/todos';
+import { SelectedStatus, Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [selectedStatus, setSelectedStatus] = useState(SelectedStatus.all);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleSetStatus = (e: React.MouseEvent<HTMLElement>) => {
+    const target = e.target as HTMLElement;
+
+    setSelectedStatus(target.textContent as SelectedStatus);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(todoData => {
+        setTodos(todoData);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+
+    return () => clearTimeout(timerId);
+  }, [errorMessage]);
+
+  function changeTodo(todoId: number) {
+    setTodos(prevTodos =>
+      prevTodos.map(todo =>
+        todo.id === todoId ? { ...todo, completed: !todo.completed } : todo,
+      ),
+    );
+  }
+
+  function addTodo(newTodo: Todo) {
```

Comment:
It is important to follow the same way of creating functions everywhere. Fix it everywhere
```suggestion
  const addTodo = (newTodo: Todo) => {
```